client/map_client.go

```diff
@@ -81,3 +81,33 @@ func (c *MapClient) GetAndVerifyMapLeaves(ctx context.Context, indexes [][]byte)
 	}
 	return leaves, nil
 }
+
+// GetAndVerifyMapLeavesByRevision verifies and returns the requested map leaves at a specific revision.
+func (c *MapClient) GetAndVerifyMapLeavesByRevision(ctx context.Context, revision int64, indexes [][]byte) ([]*trillian.MapLeaf, error) {
+	getResp, err := c.Conn.GetLeavesByRevision(ctx, &trillian.GetMapLeavesByRevisionRequest{
+		MapId:    c.MapID,
+		Index:    indexes,
+		Revision: revision,
+	})
+	if err != nil {
```

Comment:
Everything below here could be commonized with `GetAndVerifyMapLeaves`, with one small change:
```golang
// If we asked for a specific revision, check we got it.
if revision != -1 && int64(mapRoot.Revision) != revision {
...
```